test/test_fullupdate.jl

```diff
@@ -0,0 +1,76 @@
+using ITensorNetworks
+using ITensorNetworks:
+  compute_message_tensors,
+  apply_fullupdate,
+  get_environment,
+  nested_graph_leaf_vertices,
+  contract_inner
+using Test
+using Compat
+using ITensors
+using Metis
+using NamedGraphs
+using Random
+using LinearAlgebra
+using SplitApplyCombine
+
+function RandomUnitaryMatrix(N::Int)
+  x = (rand(N, N) + rand(N, N) * im) / sqrt(2)
+  f = qr(x)
+  diagR = sign.(real(diag(f.R)))
+  diagR[diagR .== 0] .= 1
+  u = f.Q * diagm(diagR)
+
+  return u
+end
+
+function ITensors.op(::OpName"RandomTwoSiteU", ::SiteType"S=1/2")
```

Comment:
There is already a `"RandomUnitary"` gate which works for any number of sites: https://github.com/ITensor/ITensors.jl/blob/v0.3.25/src/physics/site_types/generic_sites.jl